### PR TITLE
Supporting lists as a parameter value. Closes #443

### LINKED ISSUE #443: While executing taskcat test run - Unable to decode value for 'parameters: str'
**Describe the bug**
While executing "taskcat -d test run", i am getting the following error:
```
taskcat -d test run
 _            _             _   
| |_ __ _ ___| | _____ __ _| |_ 
| __/ _` / __| |/ / __/ _` | __|
| || (_| \__ \   < (_| (_| | |_ 
 \__\__,_|___/_|\_\___\__,_|\__|
                                


version 0.9.8
/home/ubuntu/.local/lib/python3.6/site-packages/dataclasses_jsonschema/__init__.py:457: UserWarning: Unable to decode value for 'parameters: str'
  warnings.warn(f"Unable to decode value for '{field}: {field_type_name}'")
[INFO   ] : Lint passed for test mytest on template /home/ubuntu/taskcatrun/Create_ALB.template
[S3: -> ] s3://tcat-taskcat-poc-nlsq599i/taskcat-poc/Create_ALB.template
[ERROR  ] : TypeError expected string or bytes-like object
Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_cli.py", line 70, in main
    cli.run()
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_cli_core.py", line 228, in run
    return getattr(command(), subcommand)(**args)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_cli_modules/test.py", line 75, in run
    parameters = config.get_rendered_parameters(buckets, regions, templates)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_config.py", line 295, in get_rendered_parameters
    region_params, s3bucket.name, region.name, region.client
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_template_params.py", line 61, in __init__
    self.transform_parameter()
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_template_params.py", line 76, in transform_parameter
    self._regex_replace_param_value(self.RE_GENNUMB, self._gen_rand_num(20))
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_template_params.py", line 348, in _regex_replace_param_value
    if self.regxfind(regex_pattern, self.param_value):
  File "/home/ubuntu/.local/lib/python3.6/site-packages/taskcat/_common_utils.py", line 75, in regxfind
    security_group = re_object.search(data_line)
TypeError: expected string or bytes-like object
ubuntu@ubuntu-VirtualBox:~/taskcatrun$ 
```

Content of .taskcat.yml is
```
project:
  name: taskcat-poc
tests:
  mytest:
    template: ./Create_ALB.template
    regions:
    - us-east-1
    parameters: 
      VPCIdnt: 'vpc-4e722d34'
      SubnetsList: 
        - 'subnet-09c2c16e'
        - 'subnet-774c9f3a'
      HealthChkPath: '/wordpress/wp-admin/install.php'
      Protocol: 'HTTP'
      PortNumber: 80
```
Please let me know, if any fix i have to make here.
taskcat version used is 0.9.8
ubuntu 18.04

_@andrew-glenn edit: Encasing traceback and config file in code blocks_

### PR DESCRIPTION
**Pull request by andrew-glenn**
This PR adds support for a list as a parameter value. Tests are included.

### CODE REVIEW

File: taskcat/_template_params.py [hunk 1]
```diff
@@ -65,6 +65,25 @@ def transform_parameter(self):
...
+                    self.region,
+                    self._boto_client,
+                    self.az_excludes,
+                )
+                nested_pg.transform_parameter()
+                for result_value in nested_pg.results.values():
+                    _results_list.append(result_value)
+                self.param_value = _results_list
+                self.results.update({param_name: _results_list})
+                return
```
Comment (jaymccon) [inline taskcat/_template_params.py]: won't this return stop additional parameters from being processed ?

Comment (andrew-glenn) [reply taskcat/_template_params.py]: Good catch. That should be a 'continue'.

**Pull request merged**